Add A* pathfinding and remove old path

ColinT opened the pull request.

ColinT committed:
Add A* pathfinding and remove old path

anson-li reviewed (changes requested):
FYI the pathing is broken at least on web, the tanks do not move. Please review the PR preview.

ColinT committed:
Replace points in path instead of replacing whole path node

ColinT committed:
More robust point_id calculation

ColinT commented:
This is now fixed. Seems like the AStar API on the JS SDK returned different point_ids than the one on desktop. I changed the calculation to be more absolute rather than rely on the Godot API.

ColinT committed:
Fix variable shadowing with p_ prefix
